### PR TITLE
Made the constructor smaller and moved if's to a array

### PR DESCRIPTION
**Pull request by itzandroidtab**
Usart0 is not implemented anymore so there will be a compiler error when someone tries to use it.

### CODE REVIEW

**Comment by itzandroidtab** [comment]
For some reason it compiled on my machine with a reinterpret_cast in a constexpr. Travis-ci was right with not compiling.

File: code/headers/uart_ports.hpp [hunk 1]
```diff
@@ -2,49 +2,13 @@
...
+
+    enum class uart_ports_c : uint8_t {
+        uart1 = 0,
+        uart2 = 1,
+        uart3 = 2,
         // only for determining how many uart ports there are
         UART_SIZE
 
     };
-
```
Comment (PatrickDekker98) [inline code/headers/uart_ports.hpp:13]: With this wip/commented code gone, shouldn't  this enum class be moved to usart_connection.hpp?

Comment (PatrickDekker98) [reply code/headers/uart_ports.hpp:13]: also uart_ports is incorect, should be usart_ports my bad

Comment (itzandroidtab) [reply code/headers/uart_ports.hpp:13]: Sure, I'm gonna move it in PR #13. And I'll change the name as well.

**Review by PatrickDekker98 — APPROVED** [review]

**Pull request merged**